app/src/ui/multi-commit-operation/choose-branch/rebase-choose-branch-dialog.tsx

```diff
@@ -157,25 +158,43 @@ export class RebaseChooseBranchDialog extends React.Component<
   private renderCleanRebaseMessage(
     currentBranch: Branch,
     baseBranch: Branch,
-    commitsToRebase: number
+    commitsAheadCount: number,
+    commitsBehindCount: number
   ) {
-    if (commitsToRebase <= 0) {
+    // The current branch is behind the base branch
+    if (commitsBehindCount > 0 && commitsAheadCount <= 0) {
+      const pluralized = commitsBehindCount === 1 ? 'commit' : 'commits'
       return (
         <>
-          This branch is up to date with{` `}
-          <strong>{currentBranch.name}</strong>
+          This will fast-forward <strong>{currentBranch.name}</strong>
+          {` to `}
+          <strong>{baseBranch.name}</strong> by applying
+          <strong>{` ${commitsBehindCount} ${pluralized}`}</strong>.
```

Comment:
Hmmm I think the `by applying X commits` shouldn't be necessary here, right? If it fast-forwards it should be clear they "become the say".

Reply:
"by applying" might be misleading, good point.
The goal was to give some more information about the consequences.

E.g., I want to fast forward to a small feature branch but choose the wrong one. A notification like "this will add 576 new commits" might save my day.

This is open for discussion though.

Reply:
One more thing to consider:
The merge dialog states the number of commits and makes no distinction between fast-forward and non-ff operation.

https://github.com/desktop/desktop/blob/development/app%2Fsrc%2Fui%2Fmulti-commit-operation%2Fchoose-branch%2Fmerge-choose-branch-dialog.tsx#L191-L201

I would vote to keep the number for reading, but rephrase the message.

Reply:
> The goal was to give some more information about the consequences.

That's a good point… maybe just `by adding X commits`? Or `including`? 🤔 

Or what about… `This will fast-forward <current-branch> X commits to match <base-branch>` ? I think this one is the most concise and no need to worry about verbs 😄

Reply:
I like the second option, sounds sound 👍

Reply:
Implemented.